File: Data Structures & Algorithms/same-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p or not q:
            return not p and not q
        left = self.isSameTree(p.left, q.left)
        right = self.isSameTree(p.right, q.right)
        return left and right and p.val == q.val

        

File: Data Structures & Algorithms/reverse-string/submission-1.kt
class Solution {
    fun reverseString(s: CharArray) {
        var left = 0
        var right = s.size - 1

        while(left <= right){
            var temp = s[left]
            s[left] = s[right]
            s[right] = temp

            left++
            right--
        }
    }
}
